python/lsst/meas/algorithms/computeExPsf.py

```diff
@@ -0,0 +1,305 @@
+                "bootstrap",
+                "marked_bootstrap",
+            ]
+        },
+    )
+
+    npatch = Field[int](
+        doc="How many patches to split the catalog into for the purpose of "
+        "jackknife variance or other options that involve running via "
+        "patches (boostrap, marked_boostrap etc.)",
+        default=1,
+        optional=True,
+    )
+
+    num_bootstrap = Field[int](
+        doc=(
+            "How many bootstrap samples to use for the 'bootstrap' and 'marked_bootstrap' var methods."
+        ),
+        default=500,
+        optional=True,
+    )
+
+    rng_seed = Field[int](
+        doc="Value to seed the treecorr random number generator with. Used to generate patches.",
+        default=13579,
+    )
+
+    def validate(self):
+        # Docs inherited from base class
+        super().validate()
+        req_params = (self.nbins, self.bin_size, self.min_sep, self.max_sep)
+        num_req_params = sum(param is not None for param in req_params)
+        if num_req_params != 3:
+            msg = (
+                "You must specify exactly three of ``nbins``, ``bin_size``, ``min_sep`` and ``max_sep``"
+                f" in treecorr_config. {num_req_params} parameters were set instead."
+            )
+            raise FieldValidationError(self.__class__.bin_size, self, msg)
+
+        if self.min_sep is not None and self.max_sep is not None:
+            if self.min_sep > self.max_sep:
+                raise FieldValidationError(
+                    self.__class__.min_sep, self, "min_sep must be <= max_sep"
+                )
+
+
+class ComputeExPsfTask(Task):
+    """Compute Ex for PSF.
+
+    Compute scalar correlation function from
+    PSF ellipticity residuals to compute TEx
+    metrics.
+
+    Parameters
+    ----------
+    de1: `np.ndarray`
+        PSF ellipticity residuals component 1.
+    de2: `np.ndarray`
+        PSF ellipticity residuals component 2.
+    ra: `np.ndarray`
+        Right ascension coordinate.
+    dec: `np.ndarray`
+        Declination coordinate.
+    units: `str`
+        In which units are ra and dec. units supported
+        are the same as the one in treecorr.
+
+    Returns
+    -------
+    struct : `lsst.pipe.base.Struct`
+            The struct contains the following data:
+        ``E1``: `float`
+            <de1 de1> scalar correlation function, compute
+            in an angular bin define in TreecorrConfig.
+        ``E2``: `float`
+            <de2 de2> scalar correlation function, compute
+            in an angular bin define in TreecorrConfig.
+        ``Ex``: `float`
+            <de1 de2> scalar cross-correlation function, compute
+            in an angular bin define in TreecorrConfig.
+    """
+
+    ConfigClass = TreecorrConfig  # ComputeExPsfConfig
+    _DefaultName = "computeExPsf"
+
+    def run(
+        self,
+        de1: npt.NDArray,
+        de2: npt.NDArray,
+        ra: npt.NDArray,
+        dec: npt.NDArray,
```

Comment:
I suspect we'd want to just take a catalog and extract these columns here. But that can wait until we start using this.